nselib/http.lua

```diff
@@ -1640,6 +1644,30 @@ function get(host, port, path, options)
     response, state = lookup_cache("GET", u.host, u.port, u.path, options);
     if ( response == nil ) then
       response = generic_request(u.host, u.port, "GET", u.path, options)
+      if(response.cookies and #response.cookies>0) then 
+        if (options and options.cookies and #options.cookies>0) then
+          --We replace the value of the cookie if same name cookie exists
+          --Else, we append it in the end.
+          local flag = 0
+          for k,v in pairs(response.cookies) do
+            for k1,v1 in pairs(options.cookies) do
+              flag = 0   
+              if(v.name == v1.name) then
+                options.cookies[k1].value = response.cookies[k].value
+                flag = 1
+                break
+              end 
+            end
+            if (flag == 0) then
+              options.cookies[#options.cookies+1] = response.cookies[k]
+            end
+          end
+        else
+          if options and options.cookies then
```

Comment:
The behavior is also different for `options.cookies` being `nil` vs. `{}` (i.e. empty).